File: app/src/main/java/com/sk/revisit/components/Component.java
package com.sk.revisit.components;

import android.content.Context;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class Component {
	protected String TAG;
	protected Context context;
	protected AppCompatActivity activity;

	Component(AppCompatActivity activity) {
		this.activity = activity;
		this.context = activity;
		this.TAG = this.getClass().getSimpleName();
	}

	public void alert(String msg) {
		Toast.makeText(context, msg, Toast.LENGTH_LONG).show();
	}
}

File: app/src/main/java/com/sk/revisit/components/Component.kt
package com.sk.revisit.components

import android.content.Context
import android.widget.Toast
import com.sk.revisit.activities.BaseActivity

open class Component(protected val activity: BaseActivity) {
    protected val tag: String = this::class.java.simpleName
    protected val context: Context = activity

    fun alert(msg: String) {
        Toast.makeText(context, msg, Toast.LENGTH_LONG).show()
    }
}